### ISSUE TITLE
ISPC exits with code 0 even if an error occured with missing folder

### ISSUE THREAD

**Issue by Lectem** [ispc/ispc#1663]
> Intel(r) SPMD Program Compiler (ispc), 1.12.0 (build commit 2a3fc60c1c60008d @ 20190816, LLVM 8.0.0)

While trying to build under linux mint, ispc does not return any error code if the output directory does not exist.

For example the command

`ispc simple.ispc -h out/simple_ispc.h -o out/simple_ispc.o && echo "should only print if successful"`
will print

> fopen: No such file or directory
> should only print if successful

if the `out` folder does not exist. It seems the behaviour is not exactly the same on windows since it successfully creates the directory if it does not exist.
I'm not sure if ispc should create the directory itself or not, but the error should at least be reported.

**Comment by dbabokin** [comment]
Good catch. I've checked how we handle result of fopen() and looks like we need some clean up.

**Comment by dbabokin** [comment]
@DeepakRajendrakumaran Please have a look.

**Comment by aneshlya** [comment]
The issue is not reproduced anymore. Currently ispc produces `Error: Cannot open output file "out/simple_ispc.o".` and exits with code 1.
